src/main/webapp/app/grading-system/grading-system.component.ts

```diff
@@ -0,0 +1,398 @@
+        this.firstPassingGrade =
+            this.gradingScale.gradeSteps.find((gradeStep) => {
+                return gradeStep.isPassingGrade;
+            })?.gradeName ?? this.gradingScale.gradeSteps[this.gradingScale.gradeSteps.length - 1].gradeName;
+    }
+
+    /**
+     * Sets the isPassingGrade property for all grade steps based on the lowerBoundInclusive property
+     * Called before a post/put request
+     *
+     * @param gradeSteps the grade steps which will be adjusted
+     */
+    setPassingGrades(gradeSteps: GradeStep[]): GradeStep[] {
+        let passingGrade = false;
+        gradeSteps.forEach((gradeStep) => {
+            if (gradeStep.gradeName === this.firstPassingGrade) {
+                passingGrade = true;
+            }
+            gradeStep.isPassingGrade = passingGrade;
+        });
+        return gradeSteps;
+    }
+
+    /**
+     * Checks if grading scale has GradeType.GRADE
+     */
+    isGradeType(): boolean {
+        return this.gradingScale.gradeType === GradeType.GRADE;
+    }
+
+    /**
+     * Create a new grade step add the end of the current grade step set
+     */
+    createGradeStep(): void {
+        const gradeStep: GradeStep = {
+            gradeName: '',
+            lowerBoundPercentage: 100,
+            upperBoundPercentage: 100,
+            isPassingGrade: true,
+            lowerBoundInclusive: this.lowerBoundInclusivity,
+            upperBoundInclusive: true,
+        };
+        if (!this.gradingScale) {
+            this.gradingScale = new GradingScale();
+        }
+        if (!this.gradingScale.gradeSteps) {
+            this.gradingScale.gradeSteps = [];
+        }
+        this.gradingScale.gradeSteps.push(gradeStep);
+    }
+
+    /**
+     * Delete grade step at given index
+     *
+     * @param index the index of the grade step
+     */
+    deleteGradeStep(index: number): void {
+        this.gradingScale.gradeSteps.splice(index, 1);
+    }
+
+    /**
+     * Generates a default grading scale to be used as template
+     */
+    generateDefaultGradingScale(): void {
+        this.gradingScale = this.getDefaultGradingScale();
+        this.firstPassingGrade = this.gradingScale.gradeSteps[3].gradeName;
+        this.lowerBoundInclusivity = true;
+    }
+
+    /**
+     * Returns the mock grading scale from the university course PSE
+     */
+    getDefaultGradingScale(): GradingScale {
+        const gradeStep1: GradeStep = {
+            gradeName: '5.0',
+            lowerBoundPercentage: 0,
+            upperBoundPercentage: 40,
+            lowerBoundInclusive: true,
+            upperBoundInclusive: false,
+            isPassingGrade: false,
+        };
+        const gradeStep2: GradeStep = {
+            gradeName: '4.7',
+            lowerBoundPercentage: 40,
+            upperBoundPercentage: 45,
+            lowerBoundInclusive: true,
+            upperBoundInclusive: false,
+            isPassingGrade: false,
+        };
```

Comment:
This function looks a bit strange with many objects inside, a function that generates them from an array of values and rules might be better. But this is just an opinion.